keps/sig-cloud-provider/2025-extend-konnectivity-for-both-directions/README.md

```diff
@@ -0,0 +1,402 @@
+         feature and to allow traffic to `localhost:6443`:
+         `--feature-gates="EnableClusterToMasterTraffic=true"`
+         `--allowed-destination=localhost:6443`.
+      3. Add the following flags to the KAS `--advertise-address=10.0.0.1`,
+         `--secure-port=6443`, and restart the process for all master nodes.
+    - In order to disable the feature you just need to remove or re-establish
+      the original `--advertise-address` flag and restart the KAS services on
+      all nodes.
+
+* **Does enabling the feature change any default behavior?**
+  Yes, all traffic originated from node network will pass from the Konnectivity
+  agents. In case network proxies are used, the agent should be configured
+  properly, and the ip used by the agents should be included in `no_proxy`
+  environment variable.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes, it can be disabled by simply changing the KAS `--advertise-address` and
+  `--secure-port` flags.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  Rolling back has no impact on subsequent re-enablements.
+
+* **Are there any tests for feature enablement/disablement?**
+  The e2e framework does not currently support enabling or disabling feature
+  gates. However, unit tests in each component dealing with managing data, created
+  with and without the feature, are necessary. At the very least, think about
+  conversion tests if API types are being modified.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g., what if some components will restart
+   mid-rollout?
+
+* **What specific metrics should inform a rollback?**
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  Describe manual testing that was done and the outcomes.
+  Longer term, we may want to require automated upgrade/rollback tests, but we
+  are missing a bunch of machinery and tooling and can't do that now.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+fields of API types, flags, etc.?**
+  Even if applying deprecation policies, they may still surprise some users.
+
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+  Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
+  checking if there are objects with field X set) may be a last resort. Avoid
+  logs or events for this purpose.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+  - [ ] Metrics
+    - Metric name:
```

Comment:
This isn't required for alpha, but if/when you move to beta, I'd like to see metrics from the konnectivity agent.